### ISSUE TITLE
Search not working: flexsearch.Document is not a constructor

### ISSUE THREAD

**Issue by llbarre** [histoire-dev/histoire#322]
### Describe the bug

When attempting to perform search, the following error message appears:

`Uncaught (in promise) TypeError: flexsearch.Document is not a constructor`

``` js
function createIndex() {
      return new flexsearch.Document({
        preset: "match",
        document: {
          id: "id",
          index: [
            "text"
          ]
        },
        worker: true,
        charset,
        language,
        tokenize: "forward"
      });
    }

```

### Reproduction

Run `story:dev`
Open up browser
Click on the Search icon
Search will not work and the message above will be in the console.

### System Info

```shell
System:
    OS: macOS 12.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 12.94 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.0 - /usr/local/bin/node
    npm: 8.3.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 106.0.5249.103
    Firefox: 105.0.1
    Safari: 16.0
  npmPackages:
    @histoire/plugin-vue: ^0.11.3 => 0.11.3 
    @vitejs/plugin-vue: ^3.1.2 => 3.1.2 
    histoire: ^0.11.3 => 0.11.3 
    vite: ^3.1.6 => 3.1.6
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by oliverzy** [comment]
It's a flexsearch issue。You can lock its version to 0.7.21

**Comment by Akryum** [comment]
Until it's fixed in flexsearch, I'm pinning it to v`0.7.21`
